contribs/gmf/src/datasource/ExternalDataSourcesManager.js

```diff
@@ -373,7 +386,17 @@ export class ExternalDatSourcesManager {
       if (wmsInfoFormat) {
         options.wmsInfoFormat = wmsInfoFormat;
       }
-      dataSource = new ngeoDatasourceOGC(options);
+
+      const legends = [];
+      if (layer.Layer) {
+        layer.Layer.forEach((layer) => {
+          legends.push(this.getLegendObject_(layer, language));
```

Comment:
It can be recursive... (Not with MapServer but with QGISserver it's the case ...)
I'm not sure that's the right way to follow, for me, we should as for standard layer build by our own the URL
`<base URL>[&?]SERVICE=WMS&VERSION=1.3.0&REQUEST=GetLegendGraphic&LAYER=<layer>&FORMAT=image/png&STYLE=default&SLD_VERSION=1.1.0`
And that's it :-)
(also without adding language)